src/main/resources/templates/fragments/generic_result_table.html

```diff
@@ -60,6 +60,21 @@ <h2>Courses</h2>
           <div th:case="'PRIMARY'">
             <td> <button type="submit" class="btn btn-primary" data-toggle="modal"
                 th:attr="data-target='#' + ${r.course.getCourseKey()}" ; id="course-detail-btn">Details</button> </td>
+                <td th:unless="${isLoggedIn}"></td>
+                <td th:if="${isLoggedIn && (r.course.classSections.indexOf(r.section) + 1 == r.course.classSections.size() 
+                                          || !r.course.classSections.get(r.course.classSections.indexOf(r.section) + 1).isSection())}">
```

Comment:
Can you explain the logic here?  It seems a bit odd.  I wonder if there is a simpler way to do this.

Reply:
Also: this is a good example of the kind of thing that should really be done in the java code and not in the view.    Instead of complicated logic like this in the view, consider adding a method on the course object called `public boolean isPrimaryWithNoSection()`.  Then just do:

```
<td th:if="${isLoggedIn && r.isPrimaryWithNoSection()}">
```

Instead of: 

```
<td th:if="${isLoggedIn && (r.course.classSections.indexOf(r.section) + 1 == r.course.classSections.size() 
                                          || !r.course.classSections.get(r.course.classSections.indexOf(r.section) + 1).isSection())}">
```
```

Reply:
Ok. I did it after trying to find a simpler way to figure out if the current r is a primary without any section, but I'll try to add a new method this time.

Reply:
The comments above the new method explains the logic of it.
I again looked into different variables each course has to solve this issue card, but I think this does the job with the two simple approaches to the purpose: checking if this lecture has no sections.